### PR TITLE
Support joining a game

### PR DESCRIPTION
**Pull request by GonSanchezS**
This pull request add functionality for a player to join an available game by updating the free player with the current player. All tests are passing.

### CODE REVIEW

**Comment by GonSanchezS** [comment]
@kellylougheed would you like to check how this works with your current code? Essentially, I'm checking which `player` has `id == 0` (which means the seat isn't taken), and updating it with the `current_player`'s ID.

File: app/models/player.rb [hunk 1]
```diff
@@ -19,4 +19,12 @@ def self.from_omniauth(auth)
       player.password = Devise.friendly_token[0,20]
     end
   end
+
+  def join_game
```
Comment (lasley) [inline app/models/player.rb]: `join_game!`

**Comment by kellylougheed** [comment]
@GonSanchezS This looks good to me so far! It looks like it should work with my code. Is the test testing the actual methods you wrote, though (the add_player method in the controller and the join_game! method)? It looks like it's just testing if attributes can be updated with the patch action, but I'm not sure if there's more going on.

**Comment by GonSanchezS** [comment]
@kellylougheed you were right. Fixed.

File: app/models/player.rb [hunk 2]
```diff
@@ -19,4 +19,14 @@ def self.from_omniauth(auth)
       player.password = Devise.friendly_token[0,20]
     end
   end
+
+  def join_game!(game)
+    if game.white_player_id == 0
+      game.update_attributes(:white_player_id => id)
+    else
+      game.update_attributes(:black_player_id => id)
```
Comment (lasley) [inline app/models/player.rb]: This is untested

File: spec/controllers/games_controller_spec.rb [hunk 3]
```diff
@@ -68,6 +68,15 @@
       game.reload
       expect(game.current_turn).to eq(1)
     end
+
+      it "should update the free player with the current player" do
+        game = FactoryGirl.create(:game, white_player_id: 0)
+        player = FactoryGirl.create(:player, email: 'meow@meow.com', password: 'MONORAILCAT')
+        sign_in player
+        player.join_game!(game)
```
Comment (GonSanchezS) [inline spec/controllers/games_controller_spec.rb]: @lasley isn't `join_game!` being tested here?

Comment (GonSanchezS) [inline spec/controllers/games_controller_spec.rb]: Got it. Added a test that checks the other side of the logic, and refactored them a bit to make it more readable.

**Pull request merged**